pre-update-user-profile-samples/pre-profile-update-validation-service-express-choreo/index.js

```diff
@@ -0,0 +1,93 @@
+const express = require('express');
+const nodemailer = require("nodemailer");
+require("dotenv").config();
+
+const app = express();
+const PORT = 3000;
+
+// Middleware to parse JSON request bodies
+app.use(express.json());
+
+// Root endpoint
+app.get('/', (req, res) => {
+    return res.status(200).json({"status": "ok", "message": "Service is running."});
+});
+
+// Mock: valid department list (simulating a directory check)
+const validDepartments = ["Engineering", "HR", "Sales", "Finance"];
+
+// Email transporter config
+const transporter = nodemailer.createTransport({
+    host: "sandbox.smtp.mailtrap.io",
+    port: 2525,
+    auth: {
+        user: process.env.MAILTRAP_USER,
+        pass: process.env.MAILTRAP_PASS
+    }
+});
+
+// Helper to extract claim values
+const getClaimValue = (claims, uri) => {
+    const claim = claims.find(c => c.uri === uri);
+    return claim ? claim.value : null;
+};
+
+app.post("/validate-user-profile-update", async (req, res) => {
+    const payload = req.body;
+
+    if (payload.actionType !== "PRE_UPDATE_PROFILE") {
+        return res.status(200).json({
+            actionStatus: "FAILED",
+            failureReason: "invalid_input",
+            failureDescription: "Invalid actionType provided."
+        });
```

Comment:
This should be an error case. Because this is not a user error that you need to show to the user
You can use the error payload and return a 500 in this case

Reply:
More over this logic is redundant as you configure the extension service to a particular extension. so ideally this should not happen, unless some external service repeats

Reply:
Remove this since it is redundant.